docs/guides/websockets/chat/javascript.md

```diff
@@ -0,0 +1,292 @@
+# Build a Multi-User Chat With JavaScript
+
+In this guide, you will build a multi-user chat application with Directus’ WebSockets interface that authenticate users with an existing account, show historical messages stored in Directus, allow users to send new messages, and immediately update all connected chats.  
+
+## Before You Start
+
+### Set Up Your Directus Project
+You will need a Directus project. If you don’t already have one, the easiest way to get started is with our [managed Directus Cloud service](https://directus.cloud). 
+
+Create a new collection called `Messages`, with a `date_created` field enabled in the *Optional System Fields* pane on collection creation. Create a text field called `text`.
+
+Create a new Role called `Users`, and create a new user with this role. Make note of the password you set.
+
+Within the `Users` role, give Create and Read access to the `Messages` collection, and Read access to the `Directus Users` system collection. 
+
+### Create a HTML Boilerplate
+
+Create an `index.html` file and open it in your code editor:
+
+```html
+<!DOCTYPE html>
+<html>
+<body>
+  <form id="login">
+    <label for="email">Email</label>
+    <input type="email" id="email">
+    <label for="password">Password</label>
+    <input type="password" id="password">
+    <input type="submit">
+  </form>
+
+  <ol></ol>
+
+  <form id="new">
+    <label for="message">Message</label>
+    <input type="text" id="text">
+    <input type="submit">
+  </form>
+
+  <script>
+    
+  </script>
+</body>
+</html>
+```
+
+The first form will handle user login and the second will handle new message submissions. The empty `<ol>` will be populated with messages. 
+
+Inside of the `<script>`, create a `url` variable being sure to replace `your-directus-url` with your project’s URL:
+
+```js
+const url = 'wss://your-directus-url/websocket'
+let socket
+```
+
+The `socket` variable will later contain a WebSocket instance.
+
+Finally, create event listeners which are triggered on the form submissions:
+
+```js
+document.querySelector('#login').onsubmit = function(event) {
+  event.preventDefault()
+}
+
+document.querySelector('#new').onsubmit = function(event) {
+  event.preventDefault()
+}
+```
+
+## Establish WebSocket Connection
+
+Within the `#login` form submit event handler, extract the `email` and `password` values from the form:
+
+```js
+const email = event.target.elements.email.value
+const password = event.target.elements.password.value
+```
+
+Create a new WebSocket, which will immediately attempt connection:
+
+```js
+socket = new WebSocket(url)
+```
+
+On connection, you must [send an authentication message before the timeout](/guides/websockets/authentication). Add an event handler for the socket’s `open` event:
+
+```js
+socket.onopen = function() {
```

Comment:
Thought about this a lot since you first said it to me. Decided that conciseness is the path, given that both are accepted syntax. It's not the job of this guide to teach people how to use event listeners generally.